src/reducers/layerEdit.js

```diff
@@ -99,8 +99,10 @@ const layerEdit = (state = null, action) => {
         case types.LAYER_EDIT_PERIOD_TYPE_SET:
             return {
                 ...state,
-                periodType: action.periodType,
-                filters: removePeriodFromFilters(state.filters),
+                periodType: action.periodType.id,
+                filters: action.clearPeriod
```

Comment:
It looks like `clearPeriod` is always the default (`true`), what's the reason for having this condition here and adding the `clearPeriod` argument to the action?

Reply:
It is set to false here: https://github.com/dhis2/maps-app/blob/feature/DHIS2-2255/src/components/periods/PeriodTypeSelect.js#L25

It was added to avoid clearing the Period dropdown when a favorite is loaded. It normally happens when the period type is changed.

Reply:
Added a comment in PeriodTypeSelect.js